src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -316,9 +316,15 @@ class WelcomePage extends Disposable {
 			}
 			const workspacesToShow = workspaces.slice(0, 5);
 			const updateEntries = async () => {
-				await this.mapListEntries(workspacesToShow, fileService, container);
+				while (ul.firstChild) {
+					ul.removeChild(ul.firstChild);
+					await this.mapListEntries(workspacesToShow, fileService, container);
```

Comment:
Why are you doing these calls in every iteration of the loop? Don't you want to just do it once after the node has been cleared?